src/Topics/StandardModernPrecision/OneClubResponses.hs

```diff
@@ -76,15 +114,66 @@ slamSingleSuit = let
         explanation fmt =
             "You've got at least mild slam interest with 12+ HCP, and a 5+ card\
           \ suit. Bid a natural " ++ output fmt (T.Bid level strain) ++ ",\
-          \ and we'll go from there. Once we find a trump fit, we'll start\
-          \ control bidding. Subsequent bids are natural 5-card (and later\
-          \ 4-card) suits, not MaFiA."
+          \ and we'll go from there. Subsequent bids are natural 5-card (and\
+          \ later 4-card) suits, not MaFiA. Once we find a trump fit, we'll\
+          \ start control bidding."
       in
         situation "Slam" action bid explanation
   in
-    smpWrapN $ return sit <~ T.allSuits
+    ( smpWrapN $ return sit <~ T.allSuits
+    , smpWrapN $ return sit <~ [T.Clubs, T.Diamonds])
+
+
+twoHeartsBalanced :: Situations
+twoHeartsBalanced = let
+    action = do
+        firstSeatOpener
+        b1C
+        oppsPass
+        withholdBid B.b1C2Halt
+    explanation fmt =
+        "You've got at least mild slam interest with 12+ HCP, but a balanced\
+      \ hand with no 5-card suit. Bid " ++ output fmt (T.Bid 2 T.Hearts) ++ "\
+      \ to show this. Partner can try " ++ output fmt (T.Bid 2 T.Notrump) ++ "\
+      \ to show hearts, " ++ output fmt (T.Bid 3 T.Clubs) ++ " as Stayman\
+      \ (regular-type! Puppet is not needed because you've already denied a\
+      \ 5-card major), or otherwise bid naturally. If partner has clubs,\
+      \ they're likely to prefer notrump, but could also try a jump to " ++
+        output fmt (T.Bid 3 T.Spades) ++ " to show that (which should be\
+      \ surprising enough for you to recognize/remember)."
+  in
+    smpWrapN . return $ situation "2HAlt" action B.b1C2Halt explanation
 
 
+oneSpadeGF :: Situations
+oneSpadeGF = let
+    explanationMin fmt =
+        "You've got game-forcing strength and a 5-card spade suit. Start with\
+      \ bidding " ++ output fmt (T.Bid 1 T.Spades) ++ " to describe this type\
+      \ of hand. Partner will bid naturally, and when we find a fit, you'll\
+      \ sign off in game to show no interest in going further (though partner\
+      \ can always push on if they've got a monster)."
+    explanationMax fmt =
+        "You've got a 5-card spade suit and at least mild slam interest. Start\
+      \ with " ++ output fmt (T.Bid 1 T.Spades) ++ " to show partner we'e at\
+      \ least game forcing. When you find a fit, start control bidding to show\
+      \ partner you're interested in slam, too."
+    sit (explanation, givenPoints) = let
+        action = do
+            firstSeatOpener
+            b1C
+            oppsPass
+            -- The compiler is concerned that we're discarding a result value on
+            -- this next line, but suggests that if it's on purpose, we can use
+            -- `_` to suppress it. The result we're discarding is the () unit.
```

Comment:
If we could give `givenPoints` an explicit type of `Action`, maybe the compiler wouldn't mind. Typing all of `sit` sounds like it's not worth it: it's, like, a `(Formatter -> String, Action) -> Direction -> Vulnerability -> Situation` or something ridiculous like that. but perhaps if we get it right, that would still be simpler than this comment?